### PR TITLE
Contrib: Update make_offlinedoc.sh

### PR DESCRIPTION
**Pull request by tamiko**
- actually download all (online) portions of the documentation
 - improve sed command

To make it short - our current offline_documentation is incomplete (to say the
least).

### CODE REVIEW

File: contrib/utilities/makeofflinedoc.sh [hunk 1]
```diff
@@ -24,19 +24,22 @@ then
...
-rm robots.txt*
-cd ..
+echo "Downloading images (this will take a long time; press ctrl-c to cancel) ..."
 
-echo "Patching html files ..."
-sed -i 's#"http://www.dealii.org/images/steps/developer/\(step-.*\)"#"images/\1"#g' step_*.html
-sed -i 's#"https://www.dealii.org/images/steps/developer/\(step-.*\)"#"images/\1"#g' step_*.html
+for i in steps/developer \
+         shape-functions/{DGPMonomial,DGPNonparametric,DGP}/P{1,2,3,4} \
+         shape-functions/{hierarchical,lagrange}/Q{1,2,3,4}; do
```
Comment (tjhei) [inline contrib/utilities/makeofflinedoc.sh]: You are hard-coding the list of folders here. The problem is that those need to be updated with every release. Wouldn't it make more sense to increase ``-l 1`` to a deeper level?

Comment (tamiko) [reply contrib/utilities/makeofflinedoc.sh]: Good point. Updated.

File: contrib/utilities/makeofflinedoc.sh [hunk 2]
```diff
@@ -24,19 +24,16 @@ then
...
 
-echo "Downloading images (press ctrl-c to cancel) ..."
-cd images
 {
-trap "echo \"(skipping)\"" SIGINT
-wget -q -nd -A svg,png,gif,webm -m -l 1 -np https://www.dealii.org/images/steps/developer/
+  trap "echo \"(skipping)\"" SIGINT
+  wget -q -nH -A svg,png,gif,webm -m -l 3 -np "https://www.dealii.org/images/steps"
+  wget -q -nH -A svg,png,gif,webm -m -l 3 -np "https://www.dealii.org/images/shape-functions"
+  rm -f robots.txt* images/robost.txt*
```
Comment (jppelteret) [inline contrib/utilities/makeofflinedoc.sh]: I'm guessing that this should be `images/robots.txt`?

Comment (tamiko) [reply contrib/utilities/makeofflinedoc.sh]: This is now fixed.

File: contrib/utilities/makeofflinedoc.sh [hunk 3]
```diff
@@ -24,19 +24,16 @@ then
   exit 1
 fi
 
-mkdir -p images
+echo "Patching html files ..."
+sed -i 's#"https\?://www.dealii.org/images/#"images/#g' step_*.html class*.html
```
Comment (bangerth) [inline contrib/utilities/makeofflinedoc.sh]: I'm coming late to the game, but is the backslash in front of the question mark a mistake? Or is this really how sed wants this?

**Comment by tamiko** [comment]
On Sat, May 19, 2018, at 09:45 CDT, Wolfgang Bangerth <notifications@github.com> wrote:

> bangerth commented on this pull request.
>
>
>
>> @@ -24,19 +24,16 @@ then
>    exit 1
>  fi
>  
> -mkdir -p images
> +echo "Patching html files ..."
> +sed -i 's#"https\?://www.dealii.org/images/#"images/#g' step_*.html class*.html
>
> I'm coming late to the game, but is the backslash in front of the question mark a mistake? Or is this really how sed wants this?

This is really how sed wants it. Omitting the backslash is a mistake.

**Comment by bangerth** [comment]
On 05/19/2018 10:55 PM, Matthias Maier wrote:
> 
> This is really how sed wants it. Omitting the backslash is a mistake.

I guess sed predates all standards on regexes :-(

I recently had to look at one of my lectures again where I'm explaining how 
archaic the syntax of 'make' is. And found that I had, back then, looked up 
how long 'make' exists (since 1976!) to realize that at the time, the number 
of programming languages to draw inspiration from was not very large. No 
wonder they had to come up with a syntax of their own -- though, really, could 
they have come up with anything more obscure?

**Comment by tamiko** [comment]
> I recently had to look at one of my lectures again where I'm explaining how 
> archaic the syntax of 'make' is. And found that I had, back then, looked up 
> how long 'make' exists (since 1976!) to realize that at the time, the number 
> of programming languages to draw inspiration from was not very large. No 
> wonder they had to come up with a syntax of their own -- though, really, could 
> they have come up with anything more obscure?

Inventing a macro substitution language to write Makefiles and shell
scripts? Wait. Someone already did and called it m4.

**Comment by bangerth** [comment]
> Inventing a macro substitution language to write Makefiles and shell
> scripts? Wait. Someone already did and called it m4.

Ain't much better, is it?

**Pull request merged**